pkg/3scale/amp/operator/apicast_reconciler.go

```diff
@@ -151,8 +157,30 @@ func (r *ApicastReconciler) apicastLogLevelEnvVarMutator(desired, existing *apps
 	return reconcilers.DeploymentConfigEnvVarReconciler(desired, existing, "APICAST_LOG_LEVEL")
 }
 
-func Apicast(apimanager *appsv1alpha1.APIManager) (*component.Apicast, error) {
-	optsProvider := NewApicastOptionsProvider(apimanager)
+func (r *ApicastReconciler) volumeMountsMutator(desired, existing *appsv1.DeploymentConfig) bool {
+	changed := false
+
+	if !reflect.DeepEqual(existing.Spec.Template.Spec.Containers[0].VolumeMounts, desired.Spec.Template.Spec.Containers[0].VolumeMounts) {
```

Comment:
Have you checked if K8s initializes some uninitialized field that could led to some issue with the comparison? Like the case with the K8s `Service`'s `ClusterIP` attribute

Reply:
The apicast operator is reconciling the volumes and volumemounts in the same way: https://github.com/3scale/apicast-operator/blob/a787307dabeef1f9b90100e55068cb25132feb9e/controllers/apps/apicast_logic_reconciler.go#L238

There is a difference, though, in the apicast operator the resource is Deployment and the 3scale operator deploys DeploymentConfigs. I have deployed and it does not seem to be anything that can ruin DeepEqual.
```
volumeMounts:
        - mountPath: /opt/app-root/src/policies/Example/0.1
          name: policy-01-example
          readOnly: true
```
```
volumes:
      - name: policy-01-example
        secret:
          defaultMode: 420
          secretName: cp-1
```

Reply:
:+1:  thanks for looking at it

Reply:
Ok, this is not entirely true. It is true for the volumemounts, however, for the volumes, the `defaultMode` is not set in the desired, hence, DeepEqual returns false:
```
 	Name: "policy-01-exampleother",
  	VolumeSource: v1.VolumeSource{
  		... // 3 identical fields
  		AWSElasticBlockStore: nil,
  		GitRepo:              nil,
  		Secret: &v1.SecretVolumeSource{
  			SecretName:  "cp-3",
  			Items:       nil,
- 			DefaultMode: nil,
+ 			DefaultMode: &420,
  			Optional:    nil,
  		},
  		NFS:   nil,
  		ISCSI: nil,
  		... // 20 identical fields
  	},
  }
```

I will write custom comparator to make it work regardless DefaultMode